servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -111,7 +173,12 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     outputStream.flush();
     outputChannel.close();
     outputStream.close();
-    mPersistedFiles.add(fileId);
+    synchronized (mPersistingFiles) {
```

Comment:
would it make sense to have a single lock for the persistence related state to avoid a possible deadlock when the locks are not acquired in a consistent order?